part3/isbn-verifier/lib/isbn_verification.rb

```diff
@@ -0,0 +1,36 @@
+def valid_isbn?(string)
+  if contains_only_valid_characters? string
+    number_only_string = string.tr("-", "")
+    conforms_to_isbn_10_formula?(number_only_string)
+  else
+    false
+  end
+end
+
+def contains_only_valid_characters?(string)
+  valid_characters = [*('0'..'9'), '-', 'X']
+  string.chars.each do |char|
+    return false unless valid_characters.include? char
+  end
+
+  true
+end
+
+def conforms_to_isbn_10_formula?(string)
+  return false if string.length != 10
+
+  numbers = string.chars.map { |char| char == "X" ? 10 : char.to_i }
+
+  (numbers[0] * 10 +
```

Comment:
This could be solved with an `each_with_index` perhaps?

Reply:
Oh yeh, it could silly me.